detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/LongParameterList.kt

```diff
@@ -62,16 +72,26 @@ class LongParameterList(
         validateConstructor(constructor)
     }
 
+    private fun KtAnnotated.isIgnored(): Boolean {
+        return annotationEntries.any {
+            ignoreAnnotated.contains(it.typeReference?.text)
+        }
+    }
+
     private fun validateConstructor(constructor: KtConstructor<*>) {
         val owner = constructor.getContainingClassOrObject()
-        val isDataClass = owner is KtClass && owner.isData()
-        if (!ignoreDataClasses || !isDataClass) {
-            validateFunction(constructor, constructorThreshold)
+        if (owner is KtClass) {
+            if (ignoreDataClasses && owner.isData()) {
+                return
+            } else if (owner.isIgnored()) {
+                return
+            }
```

Comment:
I think this code can be simplified by introducing a new function.

```suggestion
        if (owner is KtClass && owner.isDataClassOrIgnored) {
            return
        }
        
        fun KtClassOrObject.isDataClassOrIgnored() = isIgnored || ignoreDataClasses && owner.isData()
```